pkg/streamingpromql/types/data.go

```diff
@@ -27,6 +28,47 @@ type InstantVectorSeriesData struct {
 	Histograms []promql.HPoint
 }
 
+type InstantVectorSeriesDataIterator struct {
+	data   *InstantVectorSeriesData
```

Comment:
This doesn't need to be a pointer:

```suggestion
	data   InstantVectorSeriesData
```